### ISSUE TITLE
Is it possible to close a connection when no messages are received?

### ISSUE THREAD

**Issue by Jacks349** [aaugustin/websockets#836]
I have a simple use case where i'm running more tasks concurrently, every task is a different websocket connection that receives messages and is long running. Sometimes it can happen that a connection might not receive messages anymore, and in that case i would like to close the task handling that connection.  So when there haven't been messages for x second -> close the connection.

Now i just got started to this library, so i'm sorry if this question is too basic, but how can i do that? Or do i have to write my own implementation for this? I thought that using `ping_timeout` would do it, but it didn't work for me. Here is how i create a connection:

```
async def connect():
	async with websockets.client.connect('MY_URI',  ping_timeout=3) as ws:
			await ws.recv()
```

**Comment by lgrahl** [comment]
You need to combine `ping_interval` and `ping_timeout`. The maximum wait time is the sum of both.

But unless you really know you have a resource exhaustion problem (e.g. lack of file descriptors), I wouldn't reduce either of these variables and just rely on the default ping and TCP connection timeout. Reducing the interval means more packets exchanged which means more load. That needs to be carefully weighed up with the benefits (and verify that there are in fact benefits). This kind of sounds like premature optimisation. :slightly_smiling_face:

**Comment by Jacks349** [comment]
Fair enough! This is clear now, thank you a lot for the response @lgrahl !

**Comment by aaugustin** [comment]
As long as the other endpoint responds to pings, the connection will remain alive, even if no messages are exchanged.

Put a timeout on `ws.recv` and, if the timeout elapses, exit the `with connect(...)` block to close the connection.

See https://docs.python.org/3/library/asyncio-task.html#timeouts for ways to apply a timeout with asyncio.
